PRODDOCS-810: Seeds (#8849)
fishtown-analytics/docs.getdbt.com: website/docs/reference/commands/seed.md

id: "seed"
---

Seeds are configured in the `dbt_project.yml` file, where you can define options like seed file directories and column interpretation. The `dbt seed` command then loads CSV files from your project’s `seed-paths` into your <Term id="data-warehouse" />, creating relations that you can reference in downstream models.

## Overview

> This section covers common `dbt seed` outputs and options you might use when running seeds.

Author: ```suggestion
This section covers common `dbt seed` results (artifacts and CLI output) and options you might use when running seeds.
```

> - Artifacts: Running `dbt seed` produces dbt execution artifacts, including `run_results.json` for executed seeds. For details on what’s included and how to interpret it, refer to [Run results JSON file](/reference/artifacts/run-results-json).

Author: ```suggestion
Use artifacts when you want to inspect or troubleshoot what happened during execution.
- Artifacts: Running `dbt seed` produces dbt execution artifacts, including `run_results.json` for executed seeds. For details on what’s included and how to interpret it, refer to [Run results JSON file](/reference/artifacts/run-results-json).
```

id: "seed"
---

Seeds are configured in the `dbt_project.yml` file, where you can define options like seed file directories and column interpretation. The `dbt seed` command then loads CSV files from your project’s `seed-paths` into your <Term id="data-warehouse" />, creating relations that you can reference in downstream models.

> ## Details

Reviewer: we use overview generally in ref docs i think
```suggestion
## Overview
```

> This section covers common `dbt seed` outputs and options you might use when running seeds.

Reviewer: what do outputs mean here?

> - Artifacts: Running `dbt seed` produces dbt execution artifacts, including `run_results.json` for executed seeds. For details on what’s included and how to interpret it, refer to [Run results JSON file](/reference/artifacts/run-results-json).

Reviewer: i sugg having a quick intro to introduce these 3 bullets. currently it just goes right into it without prep

- Artifacts: Running `dbt seed` produces dbt execution artifacts, including `run_results.json` for executed seeds. For details on what’s included and how to interpret it, refer to [Run results JSON file](/reference/artifacts/run-results-json).

> - Selecting resources: For advanced selection patterns (tags, paths, graph operators, and more), read [Syntax overview](/reference/node-selection/syntax).

Author: ```suggestion
Use resource selection when you want to run only a subset of seeds.
- Selecting resources: For advanced selection patterns (tags, paths, graph operators, and more), read [Syntax overview](/reference/node-selection/syntax).
```

> ## Selecting seeds to run

Reviewer: ```suggestion
## Selecting seeds
```

> Specific seeds can be run using the `--select` flag to `dbt seed`. 

Reviewer: active voice
```suggestion
You can run specific seeds using the `--select` flag when executing the `dbt seed` command. For example: 
```

Reviewer: we might want to add info on what this exmaple does.

> Example output:

Reviewer: ```suggestion
```

id: "seed"
---

Seeds are configured in the `dbt_project.yml` file, where you can define options like seed file directories and column interpretation. The `dbt seed` command then loads CSV files from your project’s `seed-paths` into your <Term id="data-warehouse" />, creating relations that you can reference in downstream models.

## Overview

This section covers common `dbt seed` results (artifacts and CLI output) and options you might use when running seeds.

Use artifacts when you want to inspect or troubleshoot what happened during execution.
- Artifacts: Running `dbt seed` produces dbt execution artifacts, including `run_results.json` for executed seeds. For details on what’s included and how to interpret it, refer to [Run results JSON file](/reference/artifacts/run-results-json).
Use resource selection when you want to run only a subset of seeds.
- Selecting resources: For advanced selection patterns (tags, paths, graph operators, and more), read [Syntax overview](/reference/node-selection/syntax).

> - Flags: `dbt seed` supports dbt global flags. For the full list, refer to [Command line options](/reference/global-configs/command-line-options) and [Available flags](/reference/global-configs/about-global-configs#available-flags).

Author: ```suggestion
Use global flags when you want to control command behavior, logging, and runtime settings.
- Flags: `dbt seed` supports dbt global flags. For the full list, refer to [Command line options](/reference/global-configs/command-line-options) and [Available flags](/reference/global-configs/about-global-configs#available-flags).
```

id: "seed"
---

> Seeds are configured in the `dbt_project.yml` file, where you can define options like seed file directories and column interpretation. The `dbt seed` command then loads CSV files from your project’s `seed-paths` into your <Term id="data-warehouse" />, creating relations that you can reference in downstream models.

Reviewer: nice job Nat! i'm wondering if the seed command intro could be clearer for new users? right now, it jumps into mechanics without explaining why you'd use seeds, uses jargon like "relations," and doesn't show what you actually do.

maybe we can start with what it does first, eg.

The `dbt seed` command allows you to load static CSV files into your data warehouse as tables. This is useful for reference data (like country codes or region mappings) that you want to version control alongside your dbt project...etc.

## Overview

This section covers common `dbt seed` results (artifacts and CLI output) and options you might use when running seeds.

> Use artifacts when you want to inspect or troubleshoot what happened during execution.

Author: ```suggestion
Use artifacts when you want to inspect or troubleshoot what happened during execution:
```

- Artifacts: Running `dbt seed` produces dbt execution artifacts, including `run_results.json` for executed seeds. For details on what’s included and how to interpret it, refer to [Run results JSON file](/reference/artifacts/run-results-json).<br />

> Use resource selection when you want to run only a subset of seeds.

Author: ```suggestion
Use resource selection when you want to run only a subset of seeds:
```

- Selecting resources: For advanced selection patterns (tags, paths, graph operators, and more), read [Syntax overview](/reference/node-selection/syntax).<br />

> Use global flags when you want to control command behavior, logging, and runtime settings.

Author: ```suggestion
Use global flags when you want to control command behavior, logging, and runtime settings:
```

id: "seed"
---

> The `dbt seed` command loads static CSV files from your project’s `seed-paths` into your <Term id="data-warehouse" /> as tables. Use seeds for small, version-controlled reference datasets you want to keep alongside your project such as country codes, region mappings, or a list of business-defined categories.

Author: ```suggestion
The `dbt seed` command loads static CSV files from your project’s `seed-paths` into your <Term id="data-warehouse" /> as tables. Use seeds for small, version-controlled reference datasets you want to keep alongside your project, such as country codes, region mappings, or a list of business-defined categories.
```

14:46:16 | Finished running 1 seed in 0.14s.
```

Examples of common `dbt seed` invocations:

```shell
# Run all seeds
dbt seed

# Run one seed
dbt seed --select "country_codes"

# Rebuild selected seeds from scratch
dbt seed --select "country_codes state_codes" --full-refresh
```

> ### The --full-refresh flag

Author: ```suggestion
## The --full-refresh flag
```

Use `--full-refresh` to force a full reload of seed data (rather than an incremental update) when you need to rebuild seed tables from scratch. This is useful when:

- You want to force a clean reload of seed data:

  ```shell
  dbt seed --full-refresh
  ```

- You have changed the seed structure (for example, column names or types):

  ```shell
  dbt seed --select "country_codes" --full-refresh
  ```

- You need consistent behavior across environments after a seed change:

  ```shell
  dbt seed --select "country_codes state_codes" --full-refresh
  ```

For information about configuring seeds (for example, column types and quoting behavior), refer to [Seed configurations](/reference/seed-configs).

> ## Related content

Author: ```suggestion
## Related docs
```

14:46:16 | Finished running 1 seed in 0.14s.
```

Examples of common `dbt seed` invocations:

```shell
# Run all seeds
dbt seed

# Run one seed
dbt seed --select "country_codes"

# Rebuild selected seeds from scratch
dbt seed --select "country_codes state_codes" --full-refresh
```

> ### The --full-refresh flag

Reviewer: should this be an h3 or h2 since it's not entirely tied to select?

Use `--full-refresh` to force a full reload of seed data (rather than an incremental update) when you need to rebuild seed tables from scratch. This is useful when:

- You want to force a clean reload of seed data:

  ```shell
  dbt seed --full-refresh
  ```

- You have changed the seed structure (for example, column names or types):

  ```shell
  dbt seed --select "country_codes" --full-refresh
  ```

- You need consistent behavior across environments after a seed change:

  ```shell
  dbt seed --select "country_codes state_codes" --full-refresh
  ```

For information about configuring seeds (for example, column types and quoting behavior), refer to [Seed configurations](/reference/seed-configs).

> ## Related content

Reviewer: i think we use related docs rather than content. content is a bit formal
